### PR TITLE
feat: added support for `lts/*` syntax in nvmrc

### LINKED ISSUE #1190: Nixpacks Node.js Build Fails with .nvmrc File Containing Non-Numeric Version (e.g., lts/iron)
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug
When attempting to build a Node.js project with Nixpacks, the build fails if there is an .nvmrc file that contains a version string like `lts/iron`. It seems that the code only recognizes versions in the format `(e.g., 20.13.1)`, but nvmrc files may contain other valid formats such as `lts/iron`.

**Current behaviour:** 
The build fails because the version string in the .nvmrc file is not recognized if its a string (lts/iron) and throws following error:

```js
Syntax error: unexpected token "T_BASIC_UNESCAPED" at line 1 with value ": node version lts/iron is not valid
```

### To reproduce
1. Add an .nvmrc file with the content specifying a version as string like` lts/iron.`
2. Try to build with Nixpacks.

### Environment

Nixpacks: v1.28.1

### PR DESCRIPTION
**Pull request by shubhexists**
Fixes #1190 

PR Checklist 

- [x] Tests are added/updated if needed  
- [x] Docs are updated if needed

### CODE REVIEW

**Comment by shubhexists** [comment]
To help me better understand the codebase, I added a test for the  `to_json` function of the `BuildPlan` struct as they were not already present.
Let me know if I have to remove it as it is out of scope for this PR.

**Comment by shubhexists** [comment]
Hi @brody192, There was no mention for `.nvmrc` in the current docs in `node` provider
I added that 
Let me know if it looks fine : D

**Comment by brody192** [comment]
Awsome, thanks! looks good.

File: src/nixpacks/plan/mod.rs [hunk 1]
```diff
@@ -407,6 +407,47 @@ mod test {
         assert_eq!(result, env_plan);
     }
 
+    #[test]
+    fn test_to_json_and_from_json() {
```
Comment (coffee-cup) [inline src/nixpacks/plan/mod.rs:411]: good test 🙂

**Review by coffee-cup — APPROVED** [review]
Looks great thanks!

**Pull request merged**